### DISCUSSION TITLE
How to enable transactions when inserting or updating data.

### DISCUSSION THREAD

**Discussion by kennazeng** [MoonStorm/FastCrud#191] [Q&A]
Hi, I am a new user of FastCrud and I am currently facing a problem where I need to enable transactions when inserting and updating data. I followed the previous method, but it has not been successful. Can someone tell me how to enable transactions in FastCrud.

using (IDbConnection conn = new SqlConnection(connStr))
{
    conn.Open();
    using (IDbTransaction trans = conn.BeginTransaction())
    {
        try
        {
            foreach (var item in list)
            {
                conn.Insert<DBModel>((DBModelData)item);
            }

            trans.Commit();
        }
        catch (Exception ex)
        {
            trans.Rollback();
            conn.Close();
        }
        _logger.LogInformation("Insert success");
    }
}

**Comment by MoonStorm** [comment]
conn.Insert(item, statement => statement.AttachToTransaction(trans)). It's in the [wiki](https://github.com/MoonStorm/FastCrud/wiki#options)
